src/Http/Http.Features/src/IFormCollection.cs

```diff
@@ -77,10 +77,24 @@ public interface IFormCollection : IEnumerable<KeyValuePair<string, StringValues
     /// <exception cref="System.ArgumentNullException">
     ///     key is null.
     /// </exception>
+    /// <exception cref="System.InvalidOperationException">
+    ///     incorrect content-type.
+    /// </exception>
     /// <remarks>
+    ///     <para>
     ///     <see cref="IFormCollection" /> has a different indexer contract than
     ///     <see cref="IDictionary{TKey, TValue}" />, as it will return <c>StringValues.Empty</c> for missing entries
     ///     rather than throwing an Exception.
+    ///     </para>
+    ///     <para>
+    ///     This indexer can only be used on POST requests. Otherwise an exception of type
+    ///     <see cref="System.InvalidOperationException" /> is thrown.
+    ///     </para>
+    ///     <para>
+    ///     Invoking this property could result in thread exhaustion since it's wrapping an asynchronous implementation.
+    ///     To prevent this the method <see cref="HttpContext.Request.ReadFormAsync()" /> can be used.
```

Comment:
`ReadFormAsync()` returns this type (`IFormCollection`) at which point using this indexer is expected, so I don't think this warning belongs here. I think the warning on `HttpRequest.Form` is sufficient. We could also add it to `IFormFeature.Form`. It looks like `IFormFeature.ReadForm()` already has a similar warning, but we could update it to match.